core/integration/module_cli_test.go

```diff
@@ -1079,3 +1079,251 @@ func (m *OtherObj) FnE() *dagger.Container {
+			c := connect(ctx, t)
+
+			ctr := c.Container().
+				From(alpineImage).
+				WithMountedFile(testCLIBinPath, daggerCliFile(t, c)).
+				WithWorkdir("/work/bar").
+				With(daggerExec("init", "--sdk=go", "--name=bar", "--source=.")).
+				WithWorkdir("/work").
+				With(daggerExec("init", "--sdk=go", "--name=foo", "--source=.")).
+				With(daggerExec("install", "./bar"))
+
+			daggerjson, err := ctr.File("dagger.json").Contents(ctx)
+			require.NoError(t, err)
+			require.Contains(t, daggerjson, "bar")
+
+			daggerjson, err = ctr.
+				WithoutDirectory("/work/bar").
+				With(daggerExec("uninstall", "./bar")).
+				File("dagger.json").Contents(ctx)
+			require.NoError(t, err)
+			require.NotContains(t, daggerjson, "bar")
+		})
+	})
+
+	t.Run("git dependency", func(ctx context.Context, t *testctx.T) {
+		t.Run("uninstall a dependency configured in dagger.json with version number", func(ctx context.Context, t *testctx.T) {
+			c := connect(ctx, t)
+
+			ctr := c.Container().
+				From(alpineImage).
+				WithMountedFile(testCLIBinPath, daggerCliFile(t, c)).
+				WithWorkdir("/work").
+				With(daggerExec("init", "--sdk=go", "--name=foo", "--source=.")).
+				With(daggerExec("install", "github.com/shykes/daggerverse/hello@v0.3.0"))
+
+			daggerjson, err := ctr.File("dagger.json").Contents(ctx)
+			require.NoError(t, err)
+			require.Contains(t, daggerjson, "hello")
+
+			daggerjson, err = ctr.With(daggerExec("uninstall", "github.com/shykes/daggerverse/hello@v0.3.0")).
+				File("dagger.json").Contents(ctx)
+			require.NoError(t, err)
+			require.NotContains(t, daggerjson, "hello")
+		})
+
+		t.Run("uninstall a dependency configured in dagger.json without version number", func(ctx context.Context, t *testctx.T) {
+			c := connect(ctx, t)
+
+			ctr := c.Container().
+				From(alpineImage).
+				WithMountedFile(testCLIBinPath, daggerCliFile(t, c)).
+				WithWorkdir("/work").
+				With(daggerExec("init", "--sdk=go", "--name=foo", "--source=.")).
+				With(daggerExec("install", "github.com/shykes/daggerverse/hello"))
+
+			daggerjson, err := ctr.File("dagger.json").Contents(ctx)
+			require.NoError(t, err)
+			require.Contains(t, daggerjson, "hello")
+
+			daggerjson, err = ctr.With(daggerExec("uninstall", "github.com/shykes/daggerverse/hello@v0.3.0")).
+				File("dagger.json").Contents(ctx)
+			require.NoError(t, err)
+			require.NotContains(t, daggerjson, "hello")
+		})
+
+		t.Run("uninstall a dependency not configured in dagger.json", func(ctx context.Context, t *testctx.T) {
+			c := connect(ctx, t)
+
+			ctr := c.Container().
+				From(alpineImage).
+				WithMountedFile(testCLIBinPath, daggerCliFile(t, c)).
+				WithWorkdir("/work").
+				With(daggerExec("init", "--sdk=go", "--name=foo", "--source=."))
+
+			_, err := ctr.With(daggerExec("uninstall", "github.com/shykes/daggerverse/hello@v0.3.0")).
+				File("dagger.json").Contents(ctx)
+			require.NoError(t, err)
+		})
+
+		// this one is currently failing - I think this should be fixed
```

Comment:
Still accurate? CI checks look green to me (aside from whatever's going on with `java-dev` and `typescript-dev`)